apps/randompokedex/random_pokedex.star

```diff
@@ -0,0 +1,185 @@
+                            width = 64,
+                            height = 10,
+                            color = config.str("titleBackground", "#999"),
+                            child = render.Marquee(
+                                scroll_direction = "horizontal",
+                                width = 60,
+                                child = render.Text(
+                                    content = "# " + str(id_) + " | " + name.upper(),
+                                    color = config.str("titleForeground", "#000"),
+                                ),
+                            ),
+                        ),
+                        render.Box(
+                            # Pokemon type 1
+                            width = 42,
+                            height = 10,
+                            padding = 1,
+                            color = TYPE_COLORS[type1],
+                            child = render.Text(
+                                content = type1.upper(),
+                                color = "#000",
+                                font = "CG-pixel-3x5-mono",
+                            ),
+                        ),
+                        render.Box(
+                            # Pokemon type 2
+                            width = 42,
+                            height = 10,
+                            padding = 1,
+                            color = TYPE_COLORS[type2],
+                            child = render.Text(
+                                content = type2.upper(),
+                                color = "#000",
+                                font = "CG-pixel-3x5-mono",
+                            ),
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Color(
+                id = "titleBackground",
+                name = "Title Background",
+                desc = "Color of title background.",
+                icon = "brush",
+                default = "#999",
+            ),
+            schema.Color(
+                id = "titleForeground",
+                name = "Title Foreground",
+                desc = "Color of title foreground.",
+                icon = "brush",
+                default = "#FFF",
+            ),
+            schema.Toggle(
+                id = "shiny",
+                name = "Shiny Pokemon",
+                desc = "Show shiny variant of Pokemon",
+                icon = "star",
+                default = False,
+            ),
+        ],
+    )
+
+def get_pokemon(id):
+    url = POKEAPI_URL.format(id)
+    data = get_cachable_data(url)
+    return json.decode(data)
+
+def get_cachable_data(url):
+    key = base64.encode(url)
+
+    data = cache.get(key)
+    if data != None:
+        return base64.decode(data)
+
+    res = http.get(url, ttl_seconds = CACHE_TTL_SECONDS)
```

Comment:
Since you added `ttl_seconds` here, I should mention that you can now completely remove the cache module from the app. No need to cache the results twice.

Reply:
latest commit should address this.